web/controllers/account_controller.ex

```diff
@@ -131,14 +128,39 @@ defmodule Healthlocker.AccountController do
   def slam(conn, _params) do
     user_id = conn.assigns.current_user.id
     user = Repo.get!(User, user_id)
-    render conn, "slam.html", user: user
+    changeset = User.connect_slam(%User{})
+    render conn, "slam.html", user: user, changeset: changeset,
+                  action: account_path(conn, :check_slam)
   end
 
-  def slam_help(conn, _params) do
-    render conn, "slam_help.html"
+  def check_slam(conn, %{"user" => %{"Forename" => forename, "Surname" => surname, "NHS_Number" => nhs, "DOB" => dob}}) do
+    birthday = datetime_birthday(dob)
+    slam_user = ReadOnlyRepo.one(from e in EPJSUser,
+                where: e."Forename" == ^forename
+                and e."Surname" == ^surname
+                and e."NHS_Number" == ^nhs
+                and e."DOB" == ^birthday
+                )
+    # TODO DOB input doesn't match type in db. Removed for now, but need to fix
+    if slam_user do
+      user = Repo.get!(User, conn.assigns.current_user.id)
+      User.connect_slam(user, %{"slam_id" => slam_user."Patient_ID"})
+          |> Repo.update!
+      slam_changeset = EPJSUser.changeset(slam_user)
+      conn
+      |> put_flash(:info, "SLaM account connected!")
+      |> render("index.html", changeset: slam_changeset, user: user,
+                slam_id: slam_user.id, action: account_path(conn, :update))
+    else
+      conn
+      |> put_flash(:error, "Details do not match. Please try again later")
+      |> redirect(to: account_path(conn, :slam))
+    end
   end
 
-  def nhs_help(conn, _params) do
-    render conn, "nhs_help.html"
+  def datetime_birthday(date_string) do
```

Comment:
We should try and keep the controllers as simple as possible, could we move this to the model? Or even it's own file?